features/256-auth-gear-interaction-overview-in-next/api.md

```diff
@@ -0,0 +1,126 @@
+# auth gear interaction overview in next
+
+Base on the new product architecture decision (auth gear + cloud function and drop record gear), skygear next should bring up a new experience on how auth gear is used in cloud function and client SDK.
+
+## Goal
+
+* Keep auth gear simple, focus on auth related functionalities.
+* Keep client SDK simple, less interaction with auth gear directly.
+* More cloud functions, a developer should create corresponding cloud funtion to fulfill application logic.
+
+## Major changes
+
+* Remove admin API in client SDK.
+* Remove user profile support in auth gear.
+
+## Architecture overview
+
+![](https://i.imgur.com/GQbup7u.png)
```

Comment:
I think the "cloud function" user table is confusing. I don't think the user table in Cloud Functions is related with Skygear at all? It is just a Database for user profile and is optional and entirely controlled by users? In that case we can ignore it in our graph and just make it clear that Cloud Functions user can have their own DB.